### ISSUE TITLE
Samples fail to compile

### ISSUE THREAD

**Issue by roblem** [negativo17/cuda#2]
Just installed 2 days ago the nvidia and cuda driver/libraries in Fedora 24:
```
dnf -y install nvidia-driver nvidia-settings kernel-devel
dnf -y install cuda nvidia-driver-cuda cuda-devel cuda-samples
```
Navigating to the samples directory `/usr/share/cuda/samples/1_Utilities/deviceQuery` and attempting to make this to verify the installation gives:
```
deviceQuery.cpp:19:26: fatal error: cuda_runtime.h: No such file or directory
 #include <cuda_runtime.h>
                          ^
compilation terminated.
Makefile:250: recipe for target 'deviceQuery.o' failed
make: *** [deviceQuery.o] Error 1
```

Adding the cuda include folder to `INCLUDES` in `Makefile`:
```
INCLUDES  := -I../../common/inc,/usr/include/cuda
```
and running make, gives:
```
/usr/bin/crt/link.stub:2:26: fatal error: host_defines.h: No such file or directory
 #include "host_defines.h"
                          ^
compilation terminated.
Makefile:253: recipe for target 'deviceQuery' failed
make: *** [deviceQuery] Error 1
```
Adding the folder `/usr/include/cuda` to `LD_LIBRARY_PATH` doesn't fix the problem.

**Comment by scaronni** [comment]
> Adding the folder /usr/include/cuda to LD_LIBRARY_PATH doesn't fix the problem.

That's not a library folder, it just includes headers.

> /usr/bin/crt/link.stub:2:26: fatal error: host_defines.h: No such file or directory

That is in `/usr/include/cuda` as well, so you need to declare the folder in some other include.

I will look if I can pre-fix all the includes directives at package build time.

**Comment by scaronni** [comment]
So  a quick fix is this one:

```
cd /usr/share/cuda/samples
find . -type f -exec sed -i -e 's|/bin/nvcc|/bin/nvcc --include-path /usr/include/cuda|g' {} \;
```

Basically add everywhere the `include-path` specified. I will see if I can make it the default or at least add it to the CUDA spec file.

**Comment by scaronni** [comment]
Fixed in 8.0.44-7
